railties/lib/rails/test_unit/runner.rb

```diff
@@ -82,9 +81,17 @@ def compose_filter(runnable, filter)
 
         private
           def extract_filters(argv)
-            # Extract absolute and relative paths but skip -n /.*/ regexp filters.
+            # Extract absolute and relative paths
+            # but skip -n <filter> and --name <filter> arguments.
+
+            prev_was_name_flag = false
             argv.filter_map do |path|
-              next unless path_argument?(path)
+              if prev_was_name_flag || ["--name", "-n"].include?(path)
+                prev_was_name_flag = !prev_was_name_flag
+                next
+              end
+
+              next if path.start_with?("-")
```

Comment:
Not happy about this: a file path could possibly start with `-` (though that sounds unlikely).